dev/TeachingTip/TeachingTip.cpp

```diff
@@ -175,16 +179,31 @@ void TeachingTip::CreateLightDismissIndicatorPopup()
     }
 }
 
-void TeachingTip::UpdateBeak()
+void TeachingTip::UpdatePointer()
 {
-    auto&& beakOcclusionGrid = m_beakOcclusionGrid.get();
-    auto&& beakEdgeBorder = m_beakEdgeBorder.get();
+    // An effective placement of auto indicates that no pointer should be shown.
+    auto placement = DetermineEffectivePlacement();
+    m_currentEffectivePointerPlacementMode = placement;
+    auto&& pointerMode = PointerMode();
+    if(pointerMode == winrt::TeachingTipPointerMode::Off || (!m_target && pointerMode != winrt::TeachingTipPointerMode::On))
+    {
+        m_currentEffectivePointerPlacementMode = winrt::TeachingTipPlacementMode::Auto;
+    }
+
+    if (placement != m_currentEffectiveTipPlacementMode)
+    {
+        m_currentEffectiveTipPlacementMode = placement;
+        TeachingTipTestHooks::NotifyEffectivePlacementChanged(*this);
+    }
 
-    float height = static_cast<float>(beakOcclusionGrid.ActualHeight());
-    float width = static_cast<float>(beakOcclusionGrid.ActualWidth());
+    auto&& pointerOcclusionGrid = m_pointerOcclusionGrid.get();
+    auto&& pointerEdgeBorder = m_pointerEdgeBorder.get();
 
-    auto columnDefinitions = beakOcclusionGrid.ColumnDefinitions();
-    auto rowDefinitions = beakOcclusionGrid.RowDefinitions();
+    float height = static_cast<float>(pointerOcclusionGrid.ActualHeight());
+    float width = static_cast<float>(pointerOcclusionGrid.ActualWidth());
+
+    auto columnDefinitions = pointerOcclusionGrid.ColumnDefinitions();
+    auto rowDefinitions = pointerOcclusionGrid.RowDefinitions();
 
     float firstColumnWidth = static_cast<float>(columnDefinitions.GetAt(0).ActualWidth());
     float secondColumnWidth = static_cast<float>(columnDefinitions.GetAt(1).ActualWidth());
```

Comment:
Here are two other instances where we assume the ColumnDefinitions()'s count is large enough.